ElectronClient/app/gui/NotebookPropertiesDialog.tsx

```diff
@@ -0,0 +1,66 @@
+import * as React from 'react';
+import { useState, useEffect } from 'react';
+
+const { _ } = require('lib/locale.js');
+const { themeStyle } = require('../theme.js');
+const DialogButtonRow = require('./DialogButtonRow.min');
+
+const Folder = require('lib/models/Folder.js');
+
+interface NotebookPropertiesDialogProps {
+	theme: number,
+	folderId: string,
+	onClose: Function,
+}
+
+export default function NotebookPropertiesDialog(props: NotebookPropertiesDialogProps) {
+	console.info('Render NotebookPropertiesDialog');
+
+	const [folder, formFolder] = useState<any>({});
+
+	const theme = themeStyle(props.theme);
+
+	useEffect(() => {
+		async function fetchFolder() {
+			if (!props.folderId) {
+				formFolder({folder: null});
+			} else {
+				formFolder(await Folder.load(props.folderId));
+			}
+		}
+
+		fetchFolder();
+	}, [props.folderId]);
+
+
+	const buttonRow_click = () => {
+		props.onClose();
+	};
+
+	const saveTitle = async () => {
+		await Folder.save({id: folder.id, title: folder.title, userValidation: true});
+	};
+
+	const saveIcon = async () => {
+		await Folder.save({id: folder.id, icon: folder.icon, userValidation: true});
+	};
```

Comment:
Thanks, it looks quite good now but we don't need two separate save handlers. When the user press the Save button on the dialog, it should save both icon and title. And userValidation is the second argument.

     await Folder.save({id: folder.id, icon: folder.icon, title: folder.title }, {userValidation: true});

Reply:
I can definitely do that.

Reply:
Replaced Save Button with the fa-save icon to conform with appearance of other elements in Joplin.